src/runtime/analytics-service.js

```diff
@@ -84,21 +97,63 @@ export class AnalyticsService {
     return this.iframe_;
   }
 
+  /**
+   * @return {!string}
+   * @private
+   */
+  getQueryString_() {
+    return this.doc_.getWin().location.search.split('?')[1];
+  }
+
+  /**
+   * @return {!string}
+   * @private
+   */
+  getReferrer_() {
+    return this.doc_.getWin().document.referrer;
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  setContext_() {
+    const utmParams = parseQueryString(this.getQueryString_());
+    this.context_.setTransactionId(this.xid_.get());
+    this.context_.setReferringOrigin(this.getReferrer_());
```

Comment:
Wrap it in `parseUrl().origin` or `getOrigin()`

Reply:
done